Dockerfile

```diff
@@ -24,6 +24,8 @@ ENV LANGUAGE=en_US.UTF-8
 ENV LANG=en_US.UTF-8
 ENV LC_ALL=en_US.UTF-8
 
+RUN composer global require hirak/prestissimo
```

Comment:
nevim jestli kvuli tomu ma smyls mit dockerfilu layer

Reply:
Díky tomu, že je to extra layer tak dál jede cache všech ostatních layerů ;)

Reply:
Leda by to mohlo být u composer install - to možná dává smysl.

Reply:
jako vubec nevim k cemu je to tady dobry, co to ma cachovat? composer install se v image zavola jednou a konec, mit cache v docker layeru je afaik bad practice

Reply:
Myslel jsem obecně - když si to lokálně zbuildneš, tak ti nejspíš všechno předtím projde z cache (cache jiných projektů na php-component). A mám to tam čistě proto, že u mě trvá composer install dlouho (spojení - mám tam Waiting, Connecting, 100%). Na serveru to prakticky nikdy vidět není - tam je rovnou 100%.

Reply:
ale to bys tu cache z jinych projektu musel dostat do build contextu

Reply:
Právě že ne. Stačí, že máš cache "z $HASH1 do HASH2" a je jedno z kterého projektu to je. Takže všechny věci z php-component sdílejí stejnou cache docker layerů...

Reply:
Právě že ne. Stačí, že máš cache "z $HASH1 do HASH2" a je jedno z kterého projektu to je. Takže všechny věci z php-component sdílejí stejnou cache docker layerů...

Reply:
ne, 1. sdileji ji jen pokud maji stejny hash a tag php:7-cli muze odkazovat na ruzny hashe 
2. ta cache se naplni az commandem composer install $COMPOSER_FLAGS --no-scripts --no-autoloader ci - cili by musely mit stejnej composer.json
3. pokud by mely stejnej composer json, tak je to zase docker cache a hirak na to nema vubec zadnej vliv
4. prijde me iracionalni mit na v produkcnim imagi nejaky cache, protoze tobe se to doma pomalu buliduje

Reply:
1) Ano, což lokálně je vždy, než uděláš docker pull
2) mícháme dvě věci. Mám pocit, že ty mluvíš o composer cache a já o docker cache (všech ostatních layerů - ne composeru). Bylo to oddělené, aby fungovala docker cache na ostatní layery (apt např.). Co se týče composer cache, tak ta se skutečně dělá až tam (a je prakticky k ničemu). Ale hirak dělá hlavně to, že dělá paralelní dotazy na ty repository. Takže ti typicky stahne dependencies za ~ 1/8 času. A to se projevuje tím víc čím víc je dependencí. Tam do toho netřeba plést cache. To prostě zrychlí běh bez cache. 
3) ano
4) v produkčním imagi není žádná cache - je tam navíc čistě jen jiný algoritmus stahování závislostí. Tj. rychlejší je to jen pokud to budilduješ bez docker layer cache.

Reply:
no dobre, tak zpet na zacatek ;) prijde mi zbytecny instalovat balicek pro paralellni instalaci balicku do veci, ktera ma 5 packages a budes ji znovu lokalne buildovat nekdy za dva roky mozna nebo nikdy...

Reply:
No, v tomhle konkrétním případě to je celkem jedno - ale šlo spíš o to, že to prostě zkouším přidávat všude kde dělám (kromě connection), aby se případně ukázalo, jestli s tím není nějaký problém. 

Ve chvíli, kdy saháš někam do composeru s čímkoli, tak to poznat je. A ve větším buildu to taky poznáš. Zrovna na connection, kde jsou dva composer.json a když upravíš legacy app, tak se pouští 2x install a jeden z toho je symfony s miliardou dependencies, tak to poznat je mega. A tam si to i normálně přidávám a necommituju, pokud něco měním v buildu nebo v composeru. 

Tady je ten build ~ k ničemu. Používá se jen lokálně.